src/macros.jl

```diff
@@ -1075,51 +1096,37 @@ macro variable(args...)
             hasub && _error("Cannot specify variable upperbound twice")
             ub = esc_nonconstant(ex.args[2])
             hasub = true
-        elseif kwarg == :category
-            (t == quot(:Fixed)) && _error("Unexpected extra arguments when declaring a fixed variable")
-            t = esc_nonconstant(ex.args[2])
-            gottype = true
+        elseif kwarg == :integer
+            integer = esc_nonconstant(ex.args[2])
+        elseif kwarg == :binary
+            binary = esc_nonconstant(ex.args[2])
         else
             push!(extra_kwargs, ex)
         end
     end
 
-    if (obj !== nothing || inconstraints !== nothing || coefficients !== nothing) &&
-       (obj === nothing || inconstraints === nothing || coefficients === nothing)
-        _error("Must provide 'objective', 'inconstraints', and 'coefficients' arguments all together for column-wise modeling")
-    end
-
     sdp = any(t -> (t == :SDP), extra)
     symmetric = (sdp || any(t -> (t == :Symmetric), extra))
     extra = filter(x -> (x != :SDP && x != :Symmetric), extra) # filter out SDP and sym tag
     for ex in extra
-        if ex in var_cats
-            if t != quot(:Default) && t != quot(ex)
-                _error("A variable cannot be both of category $cat and $category. Please specify only one category.")
-            else
-                t = quot(ex)
+        if ex == :Int
+            if integer != false
+                _error("'Int' and 'integer' keyword argument cannot both be specified.")
+            end
+            integer = true
+        elseif ex == :Bin
+            if binary != false
+                _error("'Bin' and 'binary' keyword argument cannot both be specified.")
             end
+            binary = true
         end
     end
-    extra = esc.(filter(ex -> !(ex in var_cats), extra))
-
-    # Handle the column generation functionality
-    if coefficients !== nothing
-        !isa(var,Symbol) &&
-        _error("Can only create one variable at a time when adding to existing constraints.")
-
-        variablecall = :( constructvariable!($m, $(extra...), $_error, $lb, $ub, $t, $obj, $inconstraints, $coefficients, string($quotvarname), $value) )
-        addkwargs!(variablecall, extra_kwargs)
-        return assert_validmodel(m, quote
-            $variable = $variablecall
-            $(anonvar ? variable : :($escvarname = $variable))
-        end)
-    end
```

Comment:
We can have a specific function to help with column generation, but given that the information is no longer needed at the time of variable construction, I don't see a reason to keep this awkward syntax in the `@variable` macro.